src/focus-visible.js

```diff
@@ -76,37 +76,39 @@ function init() {
   }
 
   /**
-   * On `keydown`, set `hadKeyboardEvent`, add `focus-visible` class if the
-   * key was Tab/Shift-Tab or Arrow Keys.
+   * On `keydown`, disable the forceNoFocusVisible flag.
+   * This flag should be off if a user action is taking place.
    * @param {Event} e
    */
   function onKeyDown(e) {
-    var allowedKeys = [
-      9, // TAB
-      37, // LEFT
-      38, // UP
-      39, // RIGHT
-      40 // DOWN
-    ];
-
-    // Ignore keypresses if the user is holding down a modifier key.
-    if (e.altKey || e.ctrlKey || e.metaKey) {
-      return;
-    }
+    forceNoFocusVisible = false;
+  }
 
-    // Ignore keypresses which aren't related to keyboard navigation.
-    if (allowedKeys.indexOf(e.keyCode) === -1) {
-      return;
+  /**
+   * On any pointer down event (e.g. touchstart, mousedown, pointerdown),
+   * disable the forceNoFocusVisible flag.
+   * This flag should be off if a user action is taking place.
+   * Also, keep track of the clicked on element.
+   * This is useful for determining if clicking on an element calls focus() on a
+   * different element.
+   * @param {Event} e
+   */
+  function onPointerDown(e) {
+    forceNoFocusVisible = false;
+    elementToBePointerFocused = e.target;
+    // Clicking inside of a <select multiple> element will fire a mousedown
+    // on the <option> child, but focus will be set on the <select> itself.
+    if (elementToBePointerFocused.nodeName == 'OPTION') {
```

Comment:
Note. If we keep this change then we'll need to do something different for elements inside of shadow dom.

Reply:
I think what we can do there is .getRootNode() or whatever the API is called. If that returns a shadowRoot, we'll need to set this to the element's host.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Node/getRootNode